src/clr-angular/data/datagrid/providers/selection.spec.ts

```diff
@@ -250,81 +251,79 @@ export default function(): void {
 
         describe("client-side selection and pagination", function() {
             const items = [{id: 1}, {id: 2}, {id: 3}, {id: 4}, {id: 5}, {id: 6}, {id: 7}, {id: 8}, {id: 9}, {id: 10}];
-            function testSelectedItems(selectedIndexes: any[]) {
-                items.forEach((item, index) => {
+            function cloneItems() {
+                return items.map(item => {
+                    return {...item};
+                });
+            }
+            function testSelectedItems(latestItems, selectedIndexes: any[]) {
+                latestItems.forEach((item, index) => {
                     const state = selectedIndexes.indexOf(index) > -1;
                     expect(selectionInstance.isSelected(item)).toEqual(state);
                 });
             }
 
             beforeEach(function() {
-                beforeEach(function() {
-                    itemsInstance.smartenUp();
-                    itemsInstance.all = items;
-                    pageInstance.size = 3;
-                });
+                itemsInstance.smartenUp();
+                itemsInstance.all = items;
+                pageInstance.size = 3;
             });
 
             describe("multi-selection", function() {
-                function testMultiSelection() {
-                    selectionInstance.setSelected(items[2], true);
-                    testSelectedItems([2]);
-                    expect(selectionInstance.current).toEqual([items[2]]);
-                    pageInstance.current = 3;
-                    testSelectedItems([2]);
-                    expect(selectionInstance.current).toEqual([items[2]]);
-                }
-
                 beforeEach(function() {
                     selectionInstance.selectionType = SelectionType.Multi;
                 });
 
-                it("should support no trackBy", function() {
-                    testMultiSelection();
+                it("should preserve selection on page change", function() {
+                    selectionInstance.setSelected(items[2], true);
```

Comment:
Quick question about assuming defaults while testing: Should we have explicitly checked over here the initial current page value and that the item is actually not on the page after the page change? I know that the page size is 3 and you moved to page 3 so you are viewing items 6-9 but still would that additional `expect` be useful?

Reply:
Isn't that what we already test in `page.spec.ts`?

Reply:
cool. thanks!